PROTO_tests/tests/operator_test.py

```diff
@@ -0,0 +1,774 @@
+            print("\n".join(map(": ".join, dtypeerrors)))
+        print(f"  Value mismatches:         {len(valueerrors)} / {nboth}")
+        if verbose:
+            print("\n".join(map(": ".join, valueerrors)))
+        return matches == nboth
+
+    @pytest.mark.parametrize("dtype", NUMERIC_TYPES)
+    def test_pdarray_and_scalar_ops(self, dtype):
+        pda = ak.ones(100, dtype=dtype)
+        npa = np.ones(100, dtype=dtype)
+        for scal in 1, np.int64(1):
+            for ak_add, np_add in zip((pda + scal, scal + pda), (npa + scal, scal + npa)):
+                assert isinstance(ak_add, ak.pdarrayclass.pdarray)
+                assert np.allclose(ak_add.to_ndarray(), np_add)
+
+        for scal in 2, np.int64(2):
+            for ak_sub, np_sub in zip((pda - scal, scal - pda), (npa - scal, scal - npa)):
+                assert isinstance(ak_sub, ak.pdarrayclass.pdarray)
+                assert np.allclose(ak_sub.to_ndarray(), np_sub)
+
+        for scal in 5, np.int64(5):
+            for ak_mul, np_mul in zip((pda * scal, scal * pda), (npa * scal, scal * npa)):
+                assert isinstance(ak_mul, ak.pdarrayclass.pdarray)
+                assert np.allclose(ak_mul.to_ndarray(), np_mul)
+
+        if dtype not in ["bool", "uint64"]:
+            pda *= 15
+            npa *= 15
+            for scal in 3, np.int64(3):
+                for ak_div, np_div in zip((pda / scal, scal / pda), (npa / scal, scal / npa)):
+                    assert isinstance(ak_div, ak.pdarrayclass.pdarray)
+                    assert np.allclose(ak_div.to_ndarray(), np_div)
+
+    @pytest.mark.parametrize("dtype", NUMERIC_TYPES)
+    def test_concatenation(self, dtype):
+        size = 100
+        npa = make_np_arrays(size, dtype)
+        npa2 = make_np_arrays(size, dtype)
+
+        np_concat = np.concatenate([npa, npa2])
+        ak_concat = ak.concatenate([ak.array(npa), ak.array(npa2)])
+        assert 200 == len(ak_concat)
+        assert dtype == ak_concat.dtype.name
+        assert np.allclose(ak_concat.to_ndarray(), np_concat)
+
+    def test_fixed_concatenate(self):
+        for pda1, pda2 in zip(
+            (ak.arange(4), ak.linspace(0, 3, 4)), (ak.arange(4, 7), ak.linspace(4, 6, 3))
+        ):
+            ans = list(range(7))
+            assert ak.concatenate([pda1, pda2]).to_list() == ans
+            assert ak.concatenate([pda2, pda1]).to_list() == (ans[4:] + ans[:4])
+
+    def test_invert(self):
+        ak_invert = ~ak.arange(10, dtype=ak.uint64)
+        np_invert = ~np.arange(10, dtype=np.uint)
+        assert ak_invert.to_list() == np_invert.tolist()
+
+    def test_uint_bool_binops(self):
+        # Test fix for issue #1932
+        # Adding support to binopvv to correctly handle uint and bool types
+        ak_uint = ak.arange(10, dtype=ak.uint64)
+        ak_bool = ak_uint % 2 == 0
+        assert (ak_uint + ak_bool).to_list() == (ak.arange(10) + ak_bool).to_list()
+
+    def test_float_uint_binops(self):
```

Comment:
I wish we could parameterize the operator here, but I am not sure we can without a lot of headache.

Reply:
One idea I had was something like the below, but i feel like that ends up being less readable and harder to tell if we're testing the right thing

```python
for lfn, rfn in zip([`__add__`, `__sub__`, `__mul__`, ...], [`__radd__`, `__rsub__`, `__rmul__`, ...]):
    ak_ans = getattr(ak.pdarrayclass, fn)(ak_uint, akf)
    np_ans = getattr(np.whereever, fn)(np_uint, npf)
    assert np.allclose(ak_ans.to_ndarray(), np_ans, equal_nan=True)
```